src/main/java/org/terasology/worldlyTooltip/ui/WorldlyTooltip.java

```diff
@@ -128,18 +138,38 @@ public String get() {
 
         icon = find("icon", ItemIcon.class);
         if (icon != null) {
+            icon.bindIcon(new ReadOnlyBinding<TextureRegion>() {
+                @Override
+                public TextureRegion get() {
+                    if (cameraTargetSystem.isTargetAvailable()) {
+                        if (!cameraTargetSystem.isBlock()) {
+                            EntityRef targetEntity = cameraTargetSystem.getTarget();
+                            GetTooltipIconEvent getTooltipIconEvent = new GetTooltipIconEvent();
+                            targetEntity.send(getTooltipIconEvent);
```

Comment:
Yes probably can have that in the future.
Also these events are defined in the WorldlyTooltip module at present. The module needs to be added as an optional dependency (new feature- https://github.com/MovingBlocks/Terasology/pull/2767) for these events to be handled.